Test/paulmck/rcu.hpp

```diff
@@ -3,45 +3,26 @@
 #include <cstddef>
 #include <memory>
 #include <mutex>
+#include <tuple>
 #include <utility>
 
 // Derived-type approach.  All RCU-protected data structures using this
 // approach must derive from std::rcu_obj_base, which in turn derives
 // from std::rcu_head.  No idea what happens in case of multiple inheritance.
 
 namespace std {
-    template<typename T, typename D = default_delete<T>, bool E = is_empty<D>::value>
-    class rcu_obj_base: private rcu_head {
-        D deleter;
+    template<typename T, typename D = default_delete<T>>
+    class rcu_obj_base: private rcu_head, private std::tuple<D> {
```

Comment:
I tested it on clang, assuming that it would use libc++. It looks like I was wrong. When I installed libc++ and told clang to use it, the empty base class optimization stopped working.

I will try to find something else that works better.